### PR TITLE
Add editor role to the UI

### PR DESCRIPTION
**Pull request by stuartf**
- When managing the members of a collaborative document, the "Can edit" role should be available as one of the roles
- When sharing a collaborative document as a manager, the editor role should be available as one of the roles in the dropdown
- When sharing a collaborative document as an editor, the editor role should be available as one of the roles in the dropdown
- When creating a new collaborative document, all groups and users the item is shared with should become editors

### CODE REVIEW

**Comment by stuartf** [comment]
I thought editors could only give others viewer rights, are they supposed to be able to add more editors as well?

**Comment by nicolaasmatthijs** [comment]
@stuartf : Yes, viewers should be able to add both editors and viewers.

**Comment by stuartf** [comment]
OK, but they can't remove viewers and editors correct?

**Comment by nicolaasmatthijs** [comment]
@stuartf : They can only share the document (and decide whether those people should be editors or viewers in the meantime). They should not be able to update or remove roles on the document.

**Comment by stuartf** [comment]
OK, but if someone is already a viewer and the editor shares it with that person and picks editor should that be an error, or should the role get updated? What about if the person is already an editor and another editor shares it to him and picks viewer is that an error or does he lose editor status?

**Comment by nicolaasmatthijs** [comment]
What do we currently do when you share a file with a user that is already a manager? My inclination would be to just ignore the request if the role is less than that user's current role.

**Comment by stuartf** [comment]
currently if you share and search for another manager and set it to `can view` they get downgraded to a viewer, but I think it makes sense for editors to not be able to take away access only grant it.

**Comment by nicolaasmatthijs** [comment]
@stuartf : But that would only happen when the user sharing is a manager of the content, right? Or does it also do that when you're not the manager of the item?

**Comment by stuartf** [comment]
Currently as a viewer of a collabdoc I don't see a share button.

**Comment by nicolaasmatthijs** [comment]
There should be one in the clip, at least if the collabdoc is not private.

**Comment by stuartf** [comment]
Ah, that was the problem, my collabdoc was private. Viewer sharing with manager results in the manager remaining a manager.

**Comment by stuartf** [comment]
@nicolaasmatthijs I'm trying to remember what we decided to do here. IIRC it was to make editors able to share (add viewers) only, and set the default role for adding to a collabdoc back to manager. Is that right?

**Comment by stuartf** [comment]
This is ready for review

File: ui/js/content.js [hunk 1]
```diff
@@ -304,6 +304,12 @@ require(['jquery', 'underscore', 'oae.core'], function($, _, oae) {
                 'setVisibility': oae.api.content.updateContent
             }
         };
+        // collabdocs have a special editor role
```
Comment (nicolaasmatthijs) [inline ui/js/content.js]: Should start with capital case

File: ui/js/content.js [hunk 2]
```diff
@@ -304,6 +304,12 @@ require(['jquery', 'underscore', 'oae.core'], function($, _, oae) {
...
+        if (contentProfile.resourceSubType === 'collabdoc') {
+            widgetData.roles.editor = oae.api.i18n.translate('__MSG__CAN_EDIT__');
```
Comment (nicolaasmatthijs) [inline ui/js/content.js]: The editor role should be displayed in between the viewer and manager role

File: ui/js/content.js [hunk 3]
```diff
@@ -290,10 +290,10 @@ require(['jquery', 'underscore', 'oae.core'], function($, _, oae) {
      * @see manageaccess#initManageAccess
```
Comment (nicolaasmatthijs) [inline ui/js/content.js]: When sharing a collaborative doc as an editor, there should be a dropdown in the "share" overlay (similar to manage access) where you can specify the role (can edit or can view)

**Comment by nicolaasmatthijs** [comment]
Finished review. Back to @stuartf for follow-up.

**Comment by nicolaasmatthijs** [comment]
@stuartf: Would you be able to look into the follow-up for this?

File: ui/js/content.js (hunk 2 above)
Comment (stuartf) [inline ui/js/content.js]: ECMAScript standard says an Object is an `unordered collection of properties`. Looking at how we use this `roles` needs to be an array of Objects.

File: ui/js/content.js (hunk 3 above)
Comment (stuartf) [inline ui/js/content.js]: I thought we decided to have editors only able to give `viewer` roles to others and leave `manager` as the default role.

Comment (nicolaasmatthijs) [inline ui/js/content.js]: No, we decided that editors can add both other editors and viewers, so they are able to invite other people to work on the document. `manager` should indeed be the default for groups and users added during creation.

**Comment by stuartf** [comment]
ready for more review

**Comment by nicolaasmatthijs** [comment]
@stuartf : There's still some outstanding review here around sharing a document as an editor

**Comment by nicolaasmatthijs** [comment]
@stuartf: How close to completion is the outstanding follow-up?

File: ui/js/content.js [hunk 4]
```diff
@@ -290,20 +290,26 @@ require(['jquery', 'underscore', 'oae.core'], function($, _, oae) {
...
+        var widgetData = {
             'contextProfile': contentProfile,
             'messages': getManageAccessMessages(),
-            'defaultRole': contentProfile.resourceSubType === 'collabdoc' ? 'manager' : 'viewer',
-            'roles': {
-                'viewer': oae.api.i18n.translate('__MSG__CAN_VIEW__'),
-                'manager': oae.api.i18n.translate('__MSG__CAN_MANAGE__')
-            },
+            'defaultRole': 'viewer',
+            'roles': [
```
Comment (nicolaasmatthijs) [inline ui/js/content.js]: This should have been done for groups, etc. as well. I'll see if I can push a fix in time for the bash

**Comment by nicolaasmatthijs** [comment]
Followed up and merged

**Pull request merged**